src/addBusinessDays/index.ts

```diff
@@ -42,49 +42,64 @@ export default function addBusinessDays(
       ? [1, 2, 3, 4, 5]
       : options.businessDays.filter((number) => number < 7).map(toInteger)
 
-  const date = toDate(dirtyDate)
+  const initialDate = toDate(dirtyDate)
   const amount = toInteger(dirtyAmount)
   let accountedForExceptions = 0
+
   const isExcepted = (date: Date): boolean | null => {
     if (options.exceptions) {
       const exception =
         Object.keys(options.exceptions).find((e) =>
           isSameDay(new Date(e), date)
         ) || ''
-      if (options.exceptions[exception] !== undefined) {
+
+      if (options.exceptions[exception] === true) {
+        accountedForExceptions -= 1
+      } else if (options.exceptions[exception] === false) {
         accountedForExceptions += 1
       }
       return options.exceptions[exception]
     }
     return null
   }
-  const isNonWorkingDay = (date: Date) =>
-    isExcepted(date) === false || !businessDays.includes(date.getDay())
+
+  const isWorkingDay = (date: Date) => {
```

Comment:
I changed this to the positive form to avoid the double negative of `!isNonWorkingDay`